license-tool/src/main.rs

```diff
@@ -0,0 +1,378 @@
+#![deny(rust_2018_idioms)]
+#![warn(clippy::pedantic)]
+#![allow(clippy::redundant_closure_for_method_calls)]
+
+use anyhow::{bail, Context, Result};
+use argh::FromArgs;
+use regex::Regex;
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::env;
+use std::fs::{self, File};
+use std::io;
+use std::path::{Path, PathBuf};
+use url::Url;
+
+// Regex pattern to seach for ENV variables in paths
+const ENV_PATTERN: &str = r"\$(\w+).?";
+
+/// Stores arguments
+#[derive(FromArgs, PartialEq, Debug)]
+struct Args {
+    /// configuration file with the licenses to be used
+    #[argh(option, short = 'l')]
+    licenses_file: String,
+
+    #[argh(subcommand)]
+    subcommand: Subcommand,
+}
+
+/// Stores the subcommand to be executed
+#[derive(FromArgs, Debug, PartialEq)]
+#[argh(subcommand)]
+enum Subcommand {
+    SpdxId(SpdxIdArgs),
+    Path(PathArgs),
+    Fetch(FetchArgs),
+}
+
+/// Returns the spdx-id for the package
+#[derive(FromArgs, Debug, PartialEq)]
+#[argh(subcommand, name = "spdx-id")]
+struct SpdxIdArgs {
+    /// the package name used to look up for the licenses
+    #[argh(option, short = 'p')]
+    package_name: String,
+}
+
+/// Creates a copy of the licenses files in the dest directory
+#[derive(FromArgs, Debug, PartialEq)]
+#[argh(subcommand, name = "fetch")]
+struct FetchArgs {
+    /// the destination folder for the licenses
+    #[argh(option, short = 'd')]
+    destination: PathBuf,
+}
+
+/// Prints out a space-separated list of the paths to the licenses files
+#[derive(FromArgs, Debug, PartialEq)]
+#[argh(subcommand, name = "path")]
+struct PathArgs {
+    /// the package name used to look up for the licenses
+    #[argh(option, short = 'p')]
+    package_name: String,
+    /// the source folder where the licenses are
+    #[argh(option, short = 's')]
+    source: PathBuf,
+}
+
+/// Holds the configurations for package's licenses
+#[derive(Deserialize, Debug)]
+struct PackageLicense {
+    #[serde(rename(deserialize = "spdx-id"))]
+    spdx_id: String,
+    #[serde()]
+    licenses: Vec<License>,
+}
+
+/// Holds the configurations for a license
+#[derive(Deserialize, Debug, Clone)]
+struct License {
+    #[serde(rename(deserialize = "license-url"))]
+    license_url: Url,
+    #[serde()]
```

Comment:
What's this for?

Reply:
A bad engineer that doesn't know how `serde` works, I'll remove it in my next push.